Created Dio client.

rahulshamkuwar opened the pull request:
Uses Provider to make a globally avaiable Dio client. It should be used for all requests that needs the cookie, since the cookie will already be attached. Anything that doesn't need the cookie can just create a new local Dio client instead.

rahulshamkuwar committed:
Created Dio client.
Uses Provider to make a globally avaiable Dio client. It should be used for all requests that needs the cookie, since the cookie will already be attached. Anything that doesn't need the cookie can just create a new local Dio client instead.

rahulshamkuwar committed:
Merge branch 'feed-test' into api-test

rahulshamkuwar commented:
@AndrewLester I think all we need to do now is delete the inherited widget stuff?

AndrewLester commented on lib/src/services/courseApi.dart:
```diff
@@ -1,44 +1,57 @@
+        CacheConfig(
+          baseUrl: Config.baseEndpoint!,
+          defaultMaxAge: Duration(
+            days: 1,
```
Do you know if there's a way to invalidate the cache here? If not, I say we make this only ~1hr and use it solely for unauthenticated requests. That way, if someone subscribes to a new section and receives a feed item (or otherwise receives one), they'll get it instantly. It will be useful for searches though, which won't update very much.

rahulshamkuwar replied:
Yeah, we can invalidate caches, iirc. I haven't looked too deep into the caching system, but how do you want to do it?

rahulshamkuwar replied:
https://github.com/hurshi/dio-http-cache

AndrewLester replied:
Well if we only use it for section and course searching then we probably don't need it. If we end up using it for the feed though, we could invalidate the cache when a user subscribes to a section. That said, we'd probably need a more elaborate invalidation plan.

For now, I'd just change that value to like an hour.

AndrewLester commented on lib/src/services/apiService.dart:
```diff
@@ -0,0 +1,342 @@
+    }
+  }
+
+  /* void clear() {
+    super.clear();
+  }
```
This can probably be removed now right?

rahulshamkuwar replied:
Yeah, I just wanted it to exist in one commit, so if we need it in the future, I don't have to type it up again.

MattGeimer commented on lib/src/services/apiService.dart:
```diff
@@ -0,0 +1,342 @@
+    }
+  }
+
+  /* void clear() {
+    super.clear();
+  }
+
+  void close({bool force = false}) async {
+    super.close(force: force);
+  }
+
```
Does it not exist in a commit? I *really* am not a fan of leaving this much dead code

rahulshamkuwar replied:
Yeah, I deleted it in the next commit I'm going to make, just wanted a commit with this code in case we need it a later point, since it's so long

rahulshamkuwar committed:
Removed inherited widgets.
Also removed boilerplate code.
Implemented the use of AccountProvider, but havent been able to test it yet due to API changes.

rahulshamkuwar committed:
bumb version

AndrewLester commented on lib/src/providers/accountProvider.dart:
```diff
@@ -0,0 +1,162 @@
+        ),
+        data: {"course": sectionItem.fullCode, "section": sectionItem.crn});
+
+    if (response.statusCode != 200) {
```
Is there duplicated logic in these functions that could be abstracted?

rahulshamkuwar replied:
Can probably abstract the options within the class itself.

AndrewLester reviewed (approved):
All looks good, I see the library feature is now working

rahulshamkuwar committed:
Updated AccountProvider.
Added comments for more models.
Minor UI changes.
